internal/pkg/agent/protection/overlay_policy.go

```diff
@@ -0,0 +1,33 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package protection
+
+// overlayPolicy overlays/overwrites/appends the policy with the signed layer
+func overlayPolicy(policy, signed map[string]interface{}) map[string]interface{} {
```

Comment:
I think you need an explanation of how the overlay works and why it is necessary closer to the code. It isn't really clear just reading the code by itself what this does or why it is necessary.

Adding the final design to a readme file for this package would help a lot.